rust/src/dataset/optimize.rs

```diff
@@ -0,0 +1,1015 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Table maintenance for optimizing table layout.
+//!
+//! As a table is updated, it's layout can become suboptimal. For example, if
+//! a series of small streaming appends are performed, eventually there will be
+//! a large number of small files. This imposes an overhead to track the large
+//! number of files and for very small files can make it harder to read data
+//! efficiently. In this case, files can be compacted into fewer larger files.
+//!
+//! To compact files in a table, use the [compact_files] method. This currently
+//! can compact in two cases:
+//!
+//! 1. If a fragment has fewer rows than the target number of rows per fragment.
+//! 2. If a fragment has a higher percentage of deleted rows than the provided
+//!    threshold.
+//!
+//! ```rust
+//! # use std::sync::Arc;
+//! # use tokio::runtime::Runtime;
+//! # use arrow_array::{RecordBatch, RecordBatchIterator, Int64Array};
+//! # use arrow_schema::{Schema, Field, DataType};
```

Comment:
What are these `#` commented lines?

Reply:
You can hide certain details in examples behind a `#` to make the examples less verbose.